capstone/src/components/User/useSettings.jsx

```diff
@@ -0,0 +1,142 @@
+import axios from 'axios';
+import { useState, useEffect } from 'react';
+import { useToast } from '@chakra-ui/react';
+
+const baseURL = process.env.REACT_APP_BASE_URL;
+const preferenceBaseURL = '/adventure/preferences';
+const feedbackBaseURL = '/experience/feedback';
+
+function defineURL(userType) {
+  const KeysURL = ['inactive', 'active', 'info', 'update'];
+  const URL = {};
+  KeysURL.forEach((key) => {
+    if (userType === 'adventurer') {
+      URL[key] = `${baseURL}${preferenceBaseURL}/${key}`;
+    } else {
+      URL[key] = `${baseURL}${feedbackBaseURL}/${key}`;
+    }
+  });
+  return URL;
+}
+
+async function getInactiveIDs(username, URL) {
+  const values = { username };
+  return axios.post(URL.inactive, values);
+}
+
+async function getActiveIDs(username, URL) {
+  const values = { username };
+  return axios.post(URL.active, values);
+}
+
+async function getInfo(IDs, URL) {
+  const values = { IDs };
+  return axios.post(URL.info, values);
+}
+
+async function update(form, URL) {
+  return axios.post(URL.update, form);
+}
+
+// ONLY FOR ADVENTURER
+async function getPriorization(username, userType) {
+  if (userType === 'adventurer') {
+    const values = { username };
+    return axios.post(`${baseURL}${preferenceBaseURL}/prioritize`, values);
+  }
+  return { res: { data: false } };
+}
+
+export default function useSettings(userType, username) {
+  const URL = defineURL(userType);
+  const isAdventurer = userType === 'adventurer';
+  const [activeIDs, setActiveIDs] = useState([]);
+  const [inactiveIDs, setInactiveIDs] = useState([]);
+  // ONLY FOR ADVENTURER
+  const [prioritize, setPrioritize] = useState(false);
+  // ONLY FOR ADVENTURER
```

Comment:
what does this mean?

Reply:
that is for the button where Adventurers can say that the first preference should have more impact than the rest.
Since experiences are not gonna match to anything, it doesn't make sense for them to prioritize. I just added this working version, but I should probably use conditionals so that I don't create unnecessary  variables